### PR TITLE
Parametrization of `parameter_bridge` quality of service [Port of the commits (ec44770) and (86b4245) to foxy branch]

### PR DESCRIPTION
**Pull request by lucyannofrota**
Reproduces the changes made in commit https://github.com/ros2/ros1_bridge/pull/331 in ros foxy.

Parametrize Quality of Service in `parameter_bridge`

Signed-off-by: Lucyanno Frota <lucyannofrota@gmail.com>

### CODE REVIEW

**Comment by quarkytale** [comment]
Should this include `xmlrpcpp` only when ROS 1 is found as in https://github.com/ros2/ros1_bridge/pull/371 ?

**Comment by lucyannofrota** [comment]
Sure! Great suggestion @quarkytale. I already made a commit to address this port!

**Comment by quarkytale** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18734/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13260/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/19474/)

**Comment by quarkytale** [comment]
Correct CI
Linux Packaging: [Build Status](https://ci.ros2.org/job/ci_packaging_linux/543/)

**Comment by quarkytale** [comment]
Thanks for the fix. running again:
[Build Status](https://ci.ros2.org/job/ci_packaging_linux/544/)

File: include/ros1_bridge/factory.hpp
Comment (quarkytale) [inline include/ros1_bridge/factory.hpp:1]: I'm confused why these headers are empty, can they be removed?

File: include/ros1_bridge/factory_interface.hpp
Comment (quarkytale) [inline include/ros1_bridge/factory_interface.hpp:1]: Same, remove empty headers if not required

File: include/ros1_bridge/factory.hpp
Comment (lucyannofrota) [reply include/ros1_bridge/factory.hpp:1]: The header is not empty. It's just listed in the change list without any changes. (https://github.com/ros2/ros1_bridge/blob/8876252cd287150b3a0694feff331635ca7c33eb/include/ros1_bridge/factory.hpp)

I don't know a way to remove it without creating a new fork.

File: include/ros1_bridge/factory_interface.hpp
Comment (lucyannofrota) [reply include/ros1_bridge/factory_interface.hpp:1]: Same as "factory.hpp"!
The header is not empty. It's just listed in the change list without any changes. (https://github.com/ros2/ros1_bridge/blob/8876252cd287150b3a0694feff331635ca7c33eb/include/ros1_bridge/factory_interface.hpp)

I don't know a way to remove it without creating a new fork.

**Pull request merged**